### PR TITLE
bug: fix sha256 prefix handling in csi driver

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR fixes issue with the problematic code: https://github.com/Dynatrace/dynatrace-operator/blob/9a3bcd1bc6b3eb9f946d056c83474efb23dde74b/pkg/controllers/csi/provisioner/install.go#L71-L97

## How can this be tested?

`make test/e2e/cloudnative/codemodules`

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: pkg/controllers/csi/provisioner/install.go [hunk 1]
```diff
@@ -93,7 +94,9 @@ func (provisioner *OneAgentProvisioner) getDigest(dynakube dynatracev1beta1.Dyna
 		return "", err
 	}
 
-	return string(imageVersion.Digest), nil
+	digest, _ := strings.CutPrefix(string(imageVersion.Digest), "sha256:")
```
Comment (waodim) [inline pkg/controllers/csi/provisioner/install.go:97]: Would it be good to handle the case if CutPrefix returns false? Or do you think it is not necessary?

Comment (andriisoldatenko) [reply pkg/controllers/csi/provisioner/install.go:97]: it;s not necessary., 
https://go.dev/play/p/71it3wq7o_W

**Review by wepudt — COMMENTED** [review]
tests work and functionality does what is advertised.

If we would like to add more than a minimal change we could:
* separate the concerns of creating a Registry and getting the Digest in the `getDigets` function
* pass the `context.Context `from `installAgentImage`

**Comment by andriisoldatenko** [comment]
> * separate the concerns of creating a Registry and getting the Digest in the `getDigets` function

I can add context, regarding splitting code, I don't see any benefits. We use this function only once.

**Review by wepudt — APPROVED** [review]
lgtm
ran tests, looked at unit test

**Pull request closed without merging**